PaymentFactory.java

```diff
@@ -0,0 +1,17 @@
+public class PaymentFactory {
+     public Transaction getPayment(String paymentType){
+      if(paymentType == null){
+         return null;
+      }
+      if(paymentType.equalsIgnoreCase("phonepe")){
+         return new PhonePe();
+         
+      } else      if(paymentType.equalsIgnoreCase("paytm"))
+         return new Paytm();
+	return null;
+     

```

Comment:
So many If else conditions here, This could have been in more simple representation.